src/store.rs

```diff
@@ -1,18 +1,15 @@
 //! Store traits.
 use crate::block::Block;
 use crate::cid::Cid;
-use crate::codec::{Codec, Decode, Encode};
+use crate::codec::{Codec, Decode};
 use crate::error::Result;
 use crate::ipld::Ipld;
 use crate::multihash::MultihashDigest;
 use crate::path::DagPath;
 use async_trait::async_trait;
-use std::borrow::Cow;
-use std::collections::{HashSet, VecDeque};
-use std::ops::Deref;
 
 /// The store parameters.
-pub trait StoreParams: Clone + Send + Sync {
+pub trait StoreParams: Clone + Send + Sync + Unpin + 'static {
```

Comment:
This 'static appears unnecessary, am I missing a reason for it?

Reply:
prevents it needing to be propagated. there is no reason why a store params implementation requires a lifetime other than 'static.